ignite/handlers/state_param_scheduler.py

```diff
@@ -0,0 +1,377 @@
+            engine.state.param_history[self.param_name].append(value)  # type: ignore[attr-defined]
+
+    @classmethod
+    def simulate_values(cls, num_events: int, **scheduler_kwargs: Any) -> List[List[int]]:
+        """Method to simulate scheduled engine state parameter values during `num_events` events.
+
+        Args:
+            num_events: number of events during the simulation.
+            scheduler_kwargs: parameter scheduler configuration kwargs.
+
+        Returns:
+            event_index, value
+
+        Examples:
+
+        .. code-block:: python
+
+            import matplotlib.pyplot as plt
+            import numpy as np
+
+            step_state_param_values = np.array(
+                StepStateScheduler.simulate_values(
+                    num_events=20, param_name="step_scheduled_param", initial_value=10, gamma=0.99, step_size=5
+                )
+            )
+
+            plt.plot(step_state_param_values[:, 0], step_state_param_values[:, 1], label="learning rate")
+            plt.xlabel("events")
+            plt.ylabel("values")
+            plt.legend()
+
+        """
+        keys_to_remove = ["save_history"]
+        for key in keys_to_remove:
+            if key in scheduler_kwargs:
+                del scheduler_kwargs[key]
+        values = []
+        scheduler = cls(save_history=False, **scheduler_kwargs)
+        engine = Engine(lambda e, b: None)
+        for i in range(num_events):
+            scheduler(engine=engine)
+            values.append([i, getattr(engine.state, scheduler_kwargs["param_name"])])
+        return values
+
+
+class LambdaStateScheduler(StateParamScheduler):
+    """Update a parameter during training by using a user defined function.
+        User defined function is taking an event index as input and returns parameter value.
+
+    Args:
+        lambda_fn: user defined parameter update function.
+        param_name: name of parameter to update.
+        save_history: whether to log the parameter values to
+            `engine.state.param_history`, (default=False).
+
+    Examples:
+
+        .. code-block:: python
+
+            ...
+            engine = Engine(train_step)
+
+            initial_value = 10
+            gamma = 0.99
+
+            param_scheduler = LambdaStateScheduler(
+                param_name="param",
+                lambda_fn=lambda event_index: initial_value * gamma ** (event_index % 9),
+            )
+
+            param_scheduler.attach(engine, Events.EPOCH_COMPLETED)
+
+            # basic handler to print scheduled state parameter
+            engine.add_event_handler(Events.EPOCH_COMPLETED, lambda _ : print(engine.state.param))
+
+            engine.run([0] * 8, max_epochs=2)
+
+    .. versionadded:: 0.6.0
+
+    """
+
+    def __init__(self, lambda_fn: Callable, param_name: str, save_history: bool = False):
+        super(LambdaStateScheduler, self).__init__(param_name, save_history)
+        self.lambda_fn = lambda_fn
```

Comment:
A `Callable` is not pickable. This could lead to some conflicts I suppose. Did you try using `torch.save/load` ?

Reply:
You're right torch save of a ``Callable`` leads to a ``PicklingError``. Guess that a workaround would be to pass a callable class instead?